src/app/server/EnhancedSetupFlowProvider.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * This class provides access to the state of the Enhanced Setup Flow feature.
```

Comment:
I was not expecting some of the spec logic for T&C to be located within this provider - either because some logic seems like it doesn't need to be changeable, and then it it does need to be customized, I would expect to see that in a T&C provider. So it would be good to call out here what the purpose of this provider is (which logic belongs here versus in the T&C provider) and why a device maker might want to override it. Something explaining the layout - individual providers for ESF features should be for data access, while spec logic for those individual features like T&C gets put here.